pkg/stream-handler.go

```diff
@@ -62,41 +63,46 @@ func (h *streamHandler) RunStream(ctx context.Context, req *backend.RunStreamReq
 
 	for {
 		deadline, cancelFn := context.WithTimeout(context.Background(), time.Duration(30)*time.Second)
+		defer cancelFn()
```

Comment:
I am curious about the "defer"; Exactly when is that function called? And why is that "clearer" than calling it directly after the `reader.Next()` method?

Reply:
if `reader.Next()` would panic for some reason, deferred function will be called anyway.  
Deferred statement is called just before func returns, so let's say our func finishes earlier for some reason <- so we can cancel context programmatically before it timeouts - so all the functions that uses it can be aware that they can stop their work.

Btw from pulsar go-cient readme: https://github.com/apache/pulsar-client-go
they suggest to use `hasNext` as the loop condition
```golang
for reader.HasNext() {
	msg, err := reader.Next(context.Background())
```